clients/vault/src/stellar_oracle/actor.rs

```diff
@@ -0,0 +1,109 @@
+use crate::stellar_oracle::{
+    EnvelopesMap, Error, FilterTx, FilterWith, ScpMessageCollector, TxHashMap, TxSetCheckerMap, TxSetMap,
+};
+use std::collections::HashMap;
+use stellar_relay::{sdk::types::StellarMessage, StellarNodeMessage, UserControls};
+use tokio::sync::{mpsc, oneshot};
+
+pub struct ScpMessageMaps {
+    envelopes_map: EnvelopesMap,
+    txset_map: TxSetMap,
+    tx_hash_map: TxHashMap,
+}
+
+impl ScpMessageMaps {
+    fn new(collector: &ScpMessageCollector) -> Self {
+        ScpMessageMaps {
+            envelopes_map: collector.envelopes_map().clone(),
+            txset_map: collector.txset_map().clone(),
+            tx_hash_map: collector.tx_hash_map().clone(),
+        }
+    }
+
+    pub fn envelopes_map(&self) -> &EnvelopesMap {
+        &self.envelopes_map
+    }
+
+    pub fn txset_map(&self) -> &TxSetMap {
+        &self.txset_map
+    }
+
+    pub fn tx_hash_map(&self) -> &TxHashMap {
+        &self.tx_hash_map
+    }
+}
+
+pub enum Message {
+    GetCurrentMap { sender: oneshot::Sender<ScpMessageMaps> },
+}
+pub struct Actor {
+    receiver: mpsc::Receiver<Message>,
+}
+
+impl Actor {
+    pub fn new(receiver: mpsc::Receiver<Message>) -> Self {
+        Actor { receiver }
+    }
+
+    async fn handle_message(&mut self, msg: Message, collector: &ScpMessageCollector) {
+        match msg {
+            Message::GetCurrentMap { sender } => {
+                let _ = sender.send(ScpMessageMaps::new(collector));
+            }
+        };
+    }
+
+    async fn run(&mut self, mut user: UserControls, mut collector: ScpMessageCollector) -> Result<(), Error> {
```

Comment:
This is the new bit I added, where it's either going to handle messages coming from the stellar-relay, or a message coming from the event listeners, wanting to get the map.